app-kotlin/src/main/kotlin/au/com/gridstone/trainingkotlin/screens/home/HomeViewController.kt

```diff
@@ -4,57 +4,152 @@ import android.os.Bundle
+import com.google.accompanist.swiperefresh.rememberSwipeRefreshState
 import kotlinx.coroutines.launch
 import org.koin.core.component.KoinComponent
 import org.koin.core.component.get
-import kotlin.coroutines.CoroutineContext
 
-class HomeViewController : Controller(), KoinComponent, CoroutineScope {
+class HomeViewController : Controller(), KoinComponent {
 
   private val viewModel: HomeViewModel = get()
-  override val coroutineContext: CoroutineContext = Dispatchers.Main + SupervisorJob()
 
   override fun onCreateView(
     inflater: LayoutInflater,
     container: ViewGroup,
     savedViewState: Bundle?,
-  ): View = inflater.inflate(R.layout.home_view_controller, container, false)
+  ): View {
+    return ComposeView(container.context).apply {
+      setContent {
+        TrainingKotlinTheme() {
+          PokemonHomeContent(viewModel = viewModel) {
+            router.pushController(
+              RouterTransaction.with(DetailViewController(pokemon = it))
+                .popChangeHandler(HorizontalChangeHandler())
+                .pushChangeHandler(HorizontalChangeHandler())
+            )
+          }
+        }
+      }
+    }
+  }
+}
 
-  override fun onAttach(view: View) {
+@Composable
+fun PokemonHomeContent(viewModel: HomeViewModel, onSelectPokemon: (Pokemon) -> Unit) {
 
-    val presenter = HomeViewPresenter(view)
+  val state: HomeViewState by viewModel.states.collectAsState()
 
-    launch {
-      presenter.events.collect { event: HomeViewEvents ->
-        when (event) {
-          is HomeViewEvents.Refresh -> launch(Dispatchers.IO) { viewModel.getAllPokemon() }
-          is HomeViewEvents.ItemClick -> router.pushController(
-            RouterTransaction.with(DetailViewController(event.pokemon))
-              .pushChangeHandler(HorizontalChangeHandler())
-              .popChangeHandler(HorizontalChangeHandler())
-          )
-        }
+  Scaffold(
+    topBar = {
+      TopAppBar {
+        Text(
+          text = stringResource(id = R.string.pokemon),
+          style = MaterialTheme.typography.h6,
+        )
       }
     }
-
-    launch {
-      viewModel.states
-        .collect { state: HomeViewState ->
-          presenter.display(state)
+  ) { padding ->
+    SwipeRefresh(
+      modifier = Modifier
+        .fillMaxSize()
+        .padding(padding),
+      state = rememberSwipeRefreshState(isRefreshing = state == Loading),
+      onRefresh = { with(viewModel) { launch { getAllPokemon() } } }
+    ) {
+      Column(verticalArrangement = Arrangement.SpaceEvenly,
+        horizontalAlignment = Alignment.CenterHorizontally,
+        modifier = Modifier.fillMaxSize()) {
+        when (val _state: HomeViewState = state) {
+          Loading -> {
+            LoadingView()
+          }
+          Failed -> {
+            FailedView()
+          }
+          is Success -> {
+            PokemonList(_state.results.results, onSelectPokemon = onSelectPokemon)
+          }
         }
```

Comment:
no need for braces for single line expressions, as per [kotlin style guide](https://github.com/Gridstone/KotlinStyleGuide)
```kotlin
when (val _state: HomeViewState = state) {
  Loading -> LoadingView()
  Failed -> FailedView()
  is Success -> PokemonList(
     results = _state.results.results, 
     onSelectPokemon = onSelectPokemon
  )
}
```